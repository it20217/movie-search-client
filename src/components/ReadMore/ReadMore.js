import { useState, useEffect } from 'react'
import axios from 'axios'

function ReadMore(){
    const [wiki, setWiki] = useState([])
    const [omdb, setOmdb] = useState([])
    const [query, setQuery] = useState("")
    const apiKey = "43b6fe3f";
      

    function handleQuery(e){
        e.preventDefault()
        setQuery(e.target.value)
    }
    async function searchWiki(e){
        e.preventDefault()
        var responce = await axios.get('https://en.wikipedia.org/api/rest_v1/page/summary/'+query)
        setWiki(responce.data.extract)
    }
    //console.log(wiki);
    async function searchRapid(e){
        e.preventDefault()
        var responce1 = await axios.get('http://www.omdbapi.com/?t='+query+'&apikey='+apiKey)
        console.log(responce1)
        //setOmdb(responce1)
    }
    console.log(omdb);
    return(
        <div>
            <div>
                <p>Search more information on Wikipedia</p>
                <input value={query} onChange={handleQuery} placeholder="Spiderman"/>
                <button className="button" onClick={searchWiki}>Search</button>
                <p>
                    {wiki}
                </p>
                <p>Search more information on Rapid</p>
                <input value={query} onChange={handleQuery} placeholder="Spiderman"/>
                <button className="button" onClick={searchRapid}>Search</button>
                <p>
                    {wiki}
                </p>
                <p>
                    {/*rapid*/}
                </p>
            </div>
        </div>
    )
}

export default ReadMore;