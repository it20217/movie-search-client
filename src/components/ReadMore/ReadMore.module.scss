$input-border-color: #cf4f4e;
$input-border-hover: #333;
.input {
    width: 50%;
    margin-right: 10px;
    padding: 8px;
    border: 1px solid $input-border-color;
    border-radius: 3px;
    box-shadow: 5px;
    &:hover{
        border: 1px solid $input-border-hover;
    }
    &::placeholder {
        font-style: italic;
        font-size: 18px;
      }
}
.wrapper{
    background-image: url('../../assets/images/film-projector.webp');
    background-position: center top;
    background-repeat: no-repeat;
    background-size: 60%;
}