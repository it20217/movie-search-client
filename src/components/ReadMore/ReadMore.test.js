import React from 'react';
/** Snapshots rendering */
import renderer from "react-test-renderer";
/** Component */
import ReadMore from './ReadMore';

describe("ReadMore page component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<ReadMore/>).toJSON();
    expect(component).toMatchSnapshot();
  })
})