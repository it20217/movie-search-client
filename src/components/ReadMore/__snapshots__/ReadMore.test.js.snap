// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadMore page component renders correctly 1`] = `
<div
  className="flex items-start justify-center space-x-6 p-6 min-w-0 relative flex-auto"
>
  <section
    className="wrapper"
  >
    <div
      className="h-screen bg-cover bg-center flex justify-items-center items-center"
    >
      <div
        className="px-10 lg:px-32 xl:px-40"
      >
        <h1
          className="text-6xl font-semibold font-serif mb-6"
        >
          <span
            className="text-red-500"
          >
            Need more information?
          </span>
           
          <br />
          <span>
            Search below
          </span>
        </h1>
        <p
          className="text-lg max-w-md"
        >
          Just enter the movie title to see additional information from an alternative movies collection database.
        </p>
        <br />
        <input
          className="input"
          onChange={[Function]}
          placeholder="Movie title"
          value=""
        />
        <button
          className="bg-white hover:bg-gray-100 text-gray-800 font-semibold py-2 px-12 border border-gray-400 rounded shadow-md"
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </div>
  </section>
</div>
`;
